setup.py

```diff
@@ -87,10 +93,112 @@ def get_tag(self):
             python, abi = "py3", "none"
             return python, abi, plat
 
+
 except ImportError:
     bdist_wheel = None
 
 
+class CMakeExtension(Extension):
+    """Class for building a native cmake extension (C++)"""
+
+    def __init__(self, repository):
+        """Constructor of CMakeExtension class
+
+        Args:
+            repository (str): The git repository of the extension to build
+        """
+
+        name = repository.split("/")[-1]
+        super().__init__(name, sources=[])
+        self.repository = repository
+
+
+class build_ext(build_ext_orig):
+    """Class for building an  extension using cmake"""
+
+    def run(self):
+        for ext in self.extensions:
+            self.build_cmake(ext)
+        super().run()
+
+    def build_cmake(self, ext):
+
+        cwd = str(pathlib.Path().absolute())
+        build_temp = pathlib.Path(self.build_temp)
+
+        build_temp.mkdir(parents=True, exist_ok=True)
+        extdir = pathlib.Path(self.get_ext_fullpath(ext.name))
+        extdir.mkdir(parents=True, exist_ok=True)
+
+        os.chdir(str(build_temp))
+        if not os.path.isdir(ext.name):
+            self.spawn(["git", "clone", ext.repository])
+
+        os.chdir(ext.name)
+
+        if not self.dry_run:
+
+            library_name = get_library_name()
+            system = platform.system()
```

Comment:
Apart from the current py src, Darwin is supported. Do we want that? As far as I know, we're interested only in Linux and Windows. If Darwin is to be supported, we need to change the 2 if statements below to 
```
if system == "Linux":
  ...
elif system == "Windows":
  ...
if system == "Darwin":
  ...
else:
  raise OSError(message)
```